Tests/SwiftDocCTests/Infrastructure/DocumentationContext/DocumentationContextTests.swift

```diff
@@ -4256,6 +4256,92 @@ let expected = """
         problem = try XCTUnwrap(linkResolutionProblems.last)
         XCTAssertEqual(problem.diagnostic.summary, "\'NonExistingDoc\' doesn\'t exist at \'/BestBook/MyArticle\'")
     }
+    
+    func testContextRecognizesOverloads() throws {
```

Comment:
I think there could probably be a larger effort to build "mock symbol graph" functionality and replace some of the test data with that. (It would also be cool to have a test harness like the Swift compiler's lit tests or the source-compatibility suite that was able to build the symbol graphs from Swift/etc code itself, but that's a larger infrastructural effort...) Regardless, that's probably better served as its own PR or series of PRs, rather than holding up this PR.